src/server/game/Mails/ServerMailMgr.h

```diff
@@ -0,0 +1,211 @@
+/*
+ * This file is part of the AzerothCore Project. See AUTHORS file for Copyright information
+ *
+ * This program is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Affero General Public License as published by the
+ * Free Software Foundation; either version 3 of the License, or (at your
+ * option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE. See the GNU Affero General Public License for
+ * more details.
+ *
+ * You should have received a copy of the GNU General Public License along
+ * with this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+/**
+ * @file ServerMailMgr.h
+ * @brief Manages the ServerMail operations, including template loading, condition checking, and mail delivery.
+ *
+ * This class handles the loading of server mail templates, associated items, and conditions from the database.
+ * It also provides functionality to check player eligibility for receiving server mails based on configured conditions.
+ *
+ * Key features:
+ * - Supports multi-item mails via `mail_server_template_items`
+ * - Supports flexible mail conditions (level, playtime, quest, achievement) via `mail_server_template_conditions`
+ * - Ensures all related data is loaded and validated on startup
+ */
+
+#ifndef _SERVERMAILMGR_H
+#define _SERVERMAILMGR_H
+
+#include "Player.h"
```

Comment:
Imo It's not good to include a large header like this in another header, instead use class forward declaration.

Reply:
Could you explain what that is, or of there is another example in the core already. I have never done that before

Reply:
Like this:

```c++
class Player;
```

Core example: https://github.com/azerothcore/azerothcore-wotlk/blob/f818ef6876f67fa9698be088aa003cc5d6ed28be/src/server/game/Mails/Mail.h#L30

You'll have to move the CheckCondition function into the cpp file in order for it to compile, since if you forward declare a class you can't use it's class functions.